geotools/src/main/scala/geotrellis/geotools/SimpleFeatureToFeature.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016 Azavea.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.geotools
+
+import geotrellis.vector._
+
+import com.vividsolutions.jts.{geom => jts}
+import org.opengis.feature.simple.SimpleFeature
+import scala.collection._
+
+import scala.collection.JavaConverters._
+
+
+object SimpleFeatureToFeature {
+
+  private def jtsToGeotrellis(geometry: Object): Geometry = {
+    geometry match {
+      case lr: jts.LinearRing => Line(lr)
+      case ls: jts.LineString => Line(ls)
+      case pt: jts.Point => Point(pt)
+      case pg: jts.Polygon => Polygon(pg)
+      case mp: jts.MultiPoint => MultiPoint(mp)
+      case ml: jts.MultiLineString => MultiLine(ml)
+      case mp: jts.MultiPolygon => MultiPolygon(mp)
+      case gc: jts.GeometryCollection => GeometryCollection(gc)
+      case  g: jts.Geometry => throw new Exception(s"Unhandled JTS Geometry $g")
+      case _ => throw new Exception("Non-Geometry")
+    }
+  }
+
+  def apply(simpleFeature: SimpleFeature): Feature[Geometry, immutable.Map[String, Object]] = {
```

Comment:
We should have this typed on `[G <: Geometry: ClassTag]` then use a combination of

https://github.com/geotrellis/geotrellis/blob/master/vector/src/main/scala/geotrellis/vector/Geometry.scala#L81

and

https://github.com/geotrellis/geotrellis/blob/master/vector/src/main/scala/geotrellis/vector/Geometry.scala#L105

To return the correct type.

Then the `asInstanceOf` cast below gets pushed down into the moment we translate from a JTS geometry.

We could call `SimpleFeatureToFeature[Geometry]` to keep the most generic type